HIP/hip-1056.md

```diff
@@ -182,34 +185,48 @@ over the stream.
 
 #### Design for Input & Output streams
 
-The block stream items are split into two types: input and output. The input items are those known after consensus but
+The block stream items are split into two types: inputs and outputs. The input items are those known after consensus but
 before transaction execution and are the input to transaction execution. Output items are those produced during
 transaction execution. These are hashed separately into two different *block stream* merkle sub-trees. This is done so
-we can separate out a subset of historical blocks with just their input items, or just output items. That subset block stream could then be used
-for transaction replay or transaction execution verification, for example.
+we can separate out a subset of historical blocks with just their input items, or just output items. That subset block
+stream could then be used for transaction replay or transaction execution verification, for example.
 
-![Block Stream Merkle Tree](../assets/hip-blocks/block-stream-merkle.svg)
+![Block Stream Merkle Tree](../assets/hip-1056/block-stream-merkle.svg)
 
 Consensus nodes have a `state merkle tree`, which is a merkle tree containing the current state of the consensus node.
-The block stream has a separate `block stream merkle tree` which is a merkle tree of all the items in the block stream
-over the entire history of the stream. Of course, older blocks can be represented by their hash rather than the actual
-full contents, so it is possible to have a very large block stream merkle tree *in concept* that is still very efficient
-to use.
+At each block boundary, the state merkle tree is a depth two *subtree* of a `block merkle tree`. 
+The block merke tree is a binary merkle tree whose four subtrees at depth two also include the block's input items,
+the block's output items, and the merkle tree of the previous block. 
+Of course Hash computation for the block `N` merkle tree reuses the hash computed for the block `N-1` merkle tree.
+
+> Merkle Item Order vs Block Item Order: The order in which items appear in the block merkle tree are not the same 
```

Comment:
I'm quite confused by this paragraph. There seem to be two contradictory statements:

> The order in which items appear in the block merkle tree are not the same order in which are streamed

vs

> The left-to-right order of merkle leaves must be the order in which those items are encountered in the stream.

Reply:
I _believe_ the intent here is to note that items are split into multiple subtrees, and strict in-order traversal of the combined tree will not match the stream order.